extension.go

```diff
@@ -8,6 +8,50 @@ import (
 	"github.com/golang/protobuf/proto"
 )
 
+// An Extension is a custom option annotation that can be applied to an Entity to provide additional
+// semantic details and metadata about the Entity.
+type Extension interface {
+	Field
+
+	// ParentEntity returns the ParentEntity where the Extension is defined
+	DefinedIn() ParentEntity
+
+	// Extendee returns the Message that the Extension is extending
+	Extendee() Message
+
+	setExtendee(m Message)
+}
+
+type ext struct {
+	field
+
+	parent   ParentEntity
+	extendee Message
+}
+
+func (e *ext) FullyQualifiedName() string { return fullyQualifiedName(e.parent, e) }
+func (e *ext) Syntax() Syntax             { return e.parent.Syntax() }
+func (e *ext) Package() Package           { return e.parent.Package() }
+func (e *ext) File() File                 { return e.parent.File() }
+func (e *ext) BuildTarget() bool          { return e.parent.BuildTarget() }
+func (e *ext) DefinedIn() ParentEntity    { return e.parent }
+func (e *ext) Extendee() Message          { return e.extendee }
+func (e *ext) Message() Message           { return nil }
```

Comment:
I think this should return the extendee. Thoughts?

Reply:
I think it could be confusing whether `Message` is returning the extendee versus the message where the extension was defined. Although files and messages can define extensions so I suppose implicitly it should be clear because of the type. Why do you think it should return extendee?